### DISCUSSION TITLE
Github Enterprise Support

### DISCUSSION THREAD

**Discussion by EmilyRosina** [EmilyRosina/gsf-docs#3] [Seeking feedback]
### Summary

This feature allows you GSF to load for enterprise URLs.
_I have no access to enterprise URLs, so I have supplied a way to do this with some inspiration from other Chrome extension implementations._

### Access
In order to use this feature you need to:
1. Go to **options** > **advanced options**
1. Enter your `domain` under the `Github Enterprise Domains` section

### Links
- [Roadmap](https://gsf.featurebase.app/submissions/62740fd207dce3a8d1745948)

**Comment by 3kbest** [comment]
Hi @EmilyRosina . Our enterprice url to github, dont start with github., is it possible to remove the "github.". That would make it more flexible. :)

![image](https://user-images.githubusercontent.com/25566931/185129945-503be01a-4436-4f65-b36e-f5aec80c4cb5.png)

**Reply by EmilyRosina** [reply]
Hi @3kbest, thanks for the feedback! 

As I have no access to any enterprise URLs I've been coding a bit blind on those side things :sweat_smile:

This should be a quick fix and be with you shortly :+1:

**Reply by EmilyRosina** [reply]
Just released v2.0.6 with this changed for you :rocket: 
Let me know if you have any problems with it :pray:

**Reply by 3kbest** [reply]
Hi. It doesn't seem to work. Nothing happened.
![image](https://user-images.githubusercontent.com/25566931/185408050-37a4535e-be3b-4f76-bb80-06ce622bb5c2.png)

**Reply by EmilyRosina** [reply]
:thinking: That's odd.. it should work with the latest changes.
Can you please try/verify the following before I start debugging more into this? :pray: 

1. Finding `Github Saved Filters` in <a href="chrome://extensions">chrome://extensions</a> and check the `details`
1. Confirm you're running `v2.0.6`
1. Confirm the URL you want to be supported is listed under `Site Access`

![image](https://user-images.githubusercontent.com/21045073/185410220-dba4ee40-84ab-4325-8da6-8b78107e5501.png)

If the above are correct, please try the old "have you turned it off and on again" trick :sweat_smile: 
- Exit Chrome and reopen
- Refresh the Github tabs you have open

:crossed_fingers: _I'm hoping it's a caching issue, as sometimes when extensions update they don't apply until you refresh or exit/start up again._

**Reply by 3kbest** [reply]
Hmm I did try. I also tried to re-install the plugin but did not work. 
I'm using Opera browser, but should not matter I guess, for it works for the normal github.com 

![image](https://user-images.githubusercontent.com/25566931/185570525-3437c15d-e1c8-4eae-b78d-1a1461bed486.png)

**Reply by EmilyRosina** [reply]
Is there any chance your company is not using the latest [version](https://enterprise.github.com/releases) of GHE; `3.6.0`?
- [How to find version of GHE](https://stackoverflow.com/questions/25851519/as-a-user-how-can-i-determine-the-installed-version-of-github-enterprise)

I ask because I know that Github [changed the way they load their pages](https://www.reddit.com/r/github/comments/vo7qia/browser_extension_fix_after_github_made_changes/), from `pjax` to `turbo`.

---

To test if this extension _should_ work with your version of GHE, can you please copy/paste these to your dev tools console, change page and see which one logs something?

```
document.addEventListener('pjax:success', () => { console.log('pjax:success') })
document.addEventListener('turbo:load', () => { console.log('turbo:load') })
```

If `turbo:load` _does_ log to the console.. then i'm a bit stumped :thinking:

**Reply by EmilyRosina** [reply]
:wave: again @3kbest
Just to let you know, I contacted Github and they have confirmed to me that no Enterprise version currently has made the switch to use `turbo` yet; they're still using `pjax`.

What I can do, knowing that, is amend my code to support both `pjax` and `turbo` for page loading.
I am hoping there's not a case where Github uses both :sweat_smile: as that feels like it could mess a lot of things up.. but once I have that in place, it should work for you and other Enterprise users :crossed_fingers:

**Reply by EmilyRosina** [reply]
I have released `v2.0.7` with the fix (_hopefully_) to support `pjax`, i.e. all Enterprise versions :crossed_fingers:

I hope this finally gets GSF to work for you :pray:
Don't hesitate to tell me how you get on, either way.

Thanks again :wave:

**Reply by 3kbest** [reply]
Hi.

it still don't work. 😕

![image](https://user-images.githubusercontent.com/25566931/186431982-6670061c-73af-4457-a6d1-b3df17d2de69.png)
![image](https://user-images.githubusercontent.com/25566931/186432061-99224b0f-b335-40d7-b2bc-b5af7dafea1a.png)

We use GitHub Enterprise Server 3.5.3

**Reply by EmilyRosina** [reply]
Hmmm.. :thinking: everything I've implemented thus far should work, _theoretically_ (as I cannot check this myself).
Version `v2.0.7` added back-support for `pjax` and you already confirmed site access to your URL.

---

In order to get anything to log, you need to trigger GH to re-rerender the content somehow.
Try this and see :eyes: :point_down: 

1. Copy/paste [the code](https://github.com/EmilyRosina/gsf-docs/discussions/3#discussioncomment-3430300) into your console again
1. Click `Assigned`
1. You should see something logged to the console

![image](https://user-images.githubusercontent.com/21045073/186449059-15e07327-091f-46cc-b08d-d45461e7dce0.png)

---

The only other thing I can think of is that your URLs don't contain `pulls` or `issues`.
That's the _only_ other thing I ensure is true before applying GSF UI.

**Reply by 3kbest** [reply]
![image](https://user-images.githubusercontent.com/25566931/186455211-f7eb4eb3-4232-4e19-b169-585646ae76dd.png)

hmm

My url looks like this for PR and assigned tab selected
https://git.myCompanyUrl.no/pulls/assigned

**Reply by EmilyRosina** [reply]
Let's try another avenue of thinking.

Do you use any other chrome extensions for Github?
If you do, which ones work for you when using GSE?
_If their code is open source I can at least have a clue as to how they've solved this Enterprise conundrum._

**Reply by GooRoo** [reply]
I can confirm that in my case it prints `pjax:success` into the log, but the extension **does not** work. 
Unfortunately, I also do not use any other extensions to try. Is the source code for this one available?

**Reply by EmilyRosina** [reply]
Hi @GooRoo, thanks for your feedback.

I may need to remove the claim on the store page then, that it supports GH Enterprise I mean 😞 

The code is not open source, no, and I have no plans to OS it.
Sadly, I cannot access an on-prem/Enterprise version of GH to debug this either.

Sorry about this... my hands are a bit tied.

**Reply by akuzminskyi** [reply]
I have the same issue, Are there any workarounds? or how I can help to identify the issue?

**Reply by EmilyRosina** [reply]
Hey @akuzminskyi i'm kinda out of ideas tbh.. without access to GH Enterprise I cannot debug this to find out what is wrong/breaking :cry:

**Reply by akuzminskyi** [reply]
because the project is not open source, so if you want to investigate the issue further maybe to add some logs which we can share later. but accessing to GH Enterprise in not an option for most of the devs(because it's an internal tool). 
Thanks for the great projects.

**Reply by EmilyRosina** [reply]
I haven't got any logs of any kind setup.
I don't store ANY of your data at all, no tracking etc - to keep things nice and clean/easy to manage.

I assume if I start adding error tracking then it would require me to setup some kind of privacy policy?

**Reply by GooRoo** [reply]
I think it's better to create a separate build/version/extension with logs which the majority shouldn't use at all.

**Reply by EmilyRosina** [reply]
But, I surely need these logs to come from GHE users?
